py/desiutil/dust.py

```diff
@@ -0,0 +1,384 @@
+# from astropy.coordinates here.
+
+# Create rotation matrix about a given axis (x, y, z)
+
+def zrotmat(angle):
+    s = np.sin(angle)
+    c = np.cos(angle)
+    return np.array(((c, s, 0),
+                     (-s, c, 0),
+                     (0, 0, 1)))
+
+
+def yrotmat(angle):
+    s = np.sin(angle)
+    c = np.cos(angle)
+    return np.array(((c, 0, -s),
+                     (0, 1, 0),
+                     (s, 0, c)))
+
+
+def xrotmat(angle):
+    s = np.sin(angle)
+    c = np.cos(angle)
+    return np.array(((1, 0, 0),
+                     (0, c, s),
+                     (0, -s, c)))
+
+
+# constant ICRS --> FK5J2000 (See USNO Circular 179, section 3.5)
+try:
+    eta0 = np.deg2rad(-19.9 / 3600000.)
+    xi0 = np.deg2rad(9.1 / 3600000.)
+    da0 = np.deg2rad(-22.9 / 3600000.)
+    ICRS_TO_FK5J2000 = np.dot(np.dot(xrotmat(-eta0), yrotmat(xi0)), zrotmat(da0))
+except TypeError:
+    #
+    # This can happen during documentation builds.  There's no real
+    # reason to define all of this at the module level anyway.
+    #
+    eta0 = None
+    xi0 = None
+    da0 = None
+    ICRS_TO_FK5J2000 = None
+
+# FK5J2000 --> Gal
+# Note that galactic pole and zeropoint of l are somewhat arbitrary
+# and not officially defined (as far as I know). The values below are
+# from astropy.coordinates, which includes the following comment:
+# | "This gives better consistency with other codes than using the values
+# |  from Reid & Brunthaler 2004 and the best self-consistency between FK5
+# |  -> Galactic and FK5 -> FK4 -> Galactic. The lon0 angle was found by
+# |  optimizing the self-consistency."
+try:
+    ngp_fk5j2000_ra = np.deg2rad(192.8594812065348)
+    ngp_fk5j2000_dec = np.deg2rad(27.12825118085622)
+    lon0_fk5j2000 = np.deg2rad(122.9319185680026)
+    FK5J2000_TO_GAL = np.dot(np.dot(zrotmat(np.pi - lon0_fk5j2000),
+                                    yrotmat(np.pi/2. - ngp_fk5j2000_dec)),
+                             zrotmat(ngp_fk5j2000_ra))
+except TypeError:
+    #
+    # See above.
+    #
+    ngp_fk5j2000_ra = None
+    ngp_fk5j2000_dec = None
+    lon0_fk5j2000 = None
+    FK5J2000_TO_GAL = None
+
+# ICRS --> Gal: simply chain through FK5J2000
+ICRS_TO_GAL = np.dot(FK5J2000_TO_GAL, ICRS_TO_FK5J2000)
+
+
+# (lon, lat) -> [x, y, z] unit vector
+def coords2cart(lon, lat):
+    coslat = np.cos(lat)
+    return np.array((coslat * np.cos(lon), coslat * np.sin(lon), np.sin(lat)))
+
+
+# [x, y, z] unit vector -> (lon, lat)
+def cart2coords(xyz):
+    x, y, z = xyz
+    return np.arctan2(y, x), np.arctan2(z, np.sqrt(x*x + y*y))
+
+
+def _icrs_to_gal(lon, lat):
+    return cart2coords(np.dot(ICRS_TO_GAL, coords2cart(lon, lat)))
+
+
+def _fk5j2000_to_gal(lon, lat):
+    return cart2coords(np.dot(FK5J2000_TO_GAL, coords2cart(lon, lat)))
+
+
+# -----------------------------------------------------------------------------
+# bilinear interpolation (because scipy.ndimage.map_coordinates is really slow
```

Comment:
This assertion should be justified, and if it is no longer true, we should use numpy/scipy/astropy.

Reply:
I haven't actually checked that (as I have a stable and working new version of the code and don't want to mess around with it much further), but I have removed that assertion. If you feel really strongly that we should use `scipy` then I'll assess this in detail.